utils/parsable_args.py

```diff
@@ -0,0 +1,110 @@
+import argparse
+
+
+def check_for_missing_args(**kwargs):
+    missing_args = []
+    for arg, value in kwargs.items():
+        if value is None:
+            missing_args.append(arg)
+    if missing_args:
+        raise ValueError(
+            f"Missing required arguments: {', '.join(missing_args)}. "
+            "Please provide all required arguments."
+        )
+
+
+def parse_segmentation_args():
+    argparser = argparse.ArgumentParser()
+    argparser.add_argument(
+        "--patient",
+        type=str,
+        default=None,
+        help="Patient ID, e.g. 'NF0014'",
+    )
+    argparser.add_argument(
+        "--well_fov",
+        type=str,
+        default=None,
+        help="Well and field of view to process, e.g. 'A01-1'",
+    )
+    argparser.add_argument(
+        "--clip_limit",
+        type=float,
+        default=0.03,
+        help="Clip limit for contrast enhancement, default is 0.03",
+    )
+    argparser.add_argument(
+        "--twoD_method",
+        type=str,
+        default="zmax",
+        choices=["zmax", "middle"],
+        help="Method for 2D projection, either 'zmax' or 'middle'. Default is 'zmax'.",
+    )
```

Comment:
Looking at this, you have a function that checks if  all parameters have been used. You can totally remove that function and use the [required](https://docs.python.org/3/library/argparse.html#required) parameter in `argparse`.

It will raise an error if those parameters does not have an option. 

However, I do not know if the `required` parameter does exactly what your function does.